main/internal/implementations/terraform_workspace/search_repo_workspaces.go

```diff
@@ -0,0 +1,202 @@
+	filePath := fmt.Sprintf("repo/%s/%s", directory, fileName)
+	fileContent, err := os.ReadFile(filePath)
+	if err != nil {
+		return "", nil, false
+	}
+
+	workspace, err := extractWorkspaceName(ctx, filePath, fileContent)
+	if err != nil && workspace == "" {
+		return "", nil, false
+	}
+
+	details, err := extractBackendDetails(ctx, filePath, fileContent, backendType)
+	if err != nil {
+		return "", nil, false
+	}
+
+	return workspace, details, true
+}
+
+// TerraformCloudFile is a struct representation of a terraform versions.tf file.
+type TerraformCloudFile struct {
+	Cloud struct {
+		Organization string
+		Workspaces   struct {
+			Name string   `hcl:"name,optional"`
+			Tags []string `hcl:"tags,optional"`
+		} `hcl:"workspaces,block"`
+	} `hcl:"cloud,block"`
+}
+
+// extractWorkspaceName extracts the workspace name from a .tf file if it exists.
+func extractWorkspaceName(ctx context.Context, filePath string, fileContent []byte) (string, error) {
+	var config TerraformCloudFile
+	err := hclsimple.Decode(filePath, fileContent, nil, &config)
+	if err != nil {
+		return "", err
+	}
+
+	return config.Cloud.Workspaces.Name, nil
+}
+
+// TerraformBackendS3 is a struct representation of a terraform backend.tf file for aws provider.
+type TerraformBackendS3 struct {
+	BackendS3 BackendS3 `hcl:"backend,s3"`
+}
+
+// BackendS3 is a struct representation of a terraform backend.tf file for s3 backend.
+type BackendS3 struct {
+	Bucket string `hcl:"bucket,optional"`
+	Key    string `hcl:"key,optional"`
+	Region string `hcl:"region,optional"`
+}
+
+// TerraformBackendAzure is a struct representation of a terraform backend.tf file for azure provider.
+type TerraformBackendAzure struct {
+	BackendAzure BackendAzure `hcl:"backend,azurerm"`
+}
+
+// BackendAzure is a struct representation of a terraform backend.tf file for azure blob storage.
+type BackendAzure struct {
+	ResourceGroupName  string `hcl:"resource_group_name,optional"`
+	StorageAccountName string `hcl:"storage_account_name,optional"`
+	ContainerName      string `hcl:"container_name,optional"`
+	Key                string `hcl:"key,optional"`
+}
+
+// TerraformBackendGoogle is a struct representation of a terraform backend.tf file for google provider.
+type TerraformBackendGoogle struct {
+	BackendGCS BackendGCS `hcl:"backend,gcs"`
+}
+
+// BackendGCS is a struct representation of a terraform backend.tf file for gcs.
+type BackendGCS struct {
+	Bucket string `hcl:"bucket,optional"`
+	Prefix string `hcl:"prefix,optional"`
+}
+
+// extractBackendDetails extracts the backend details from a .tf file if it exists.
+func extractBackendDetails(ctx context.Context, filePath string, fileContent []byte, backendType string) (interface{}, error) {
+	switch backendType {
+	case "s3":
+		var config TerraformBackendS3
+		err := hclsimple.Decode(filePath, fileContent, nil, &config)
+		if err != nil {
+			return "", err
+		}
+
+		return config.BackendS3, nil
+	case "azure_blob":
```

Comment:
All references to any azure backend should be renamed to `azurerm` to be consistent with Terraform naming conventions: https://developer.hashicorp.com/terraform/language/settings/backends/azurerm